include/boost/json/impl/object.ipp

```diff
@@ -556,6 +546,49 @@ erase(string_view key) noexcept ->
     return 1;
 }
 
+auto
+object::
+stable_erase(const_iterator pos) noexcept ->
```

Comment:
`stable_erase` and `erase` share quite a lot of code. A feel like a better approach is having a private  function template `iterator do_erase(const_iterator pos, F get_pb)` that ends with
```c++
auto result = p;
while( p != end() )
{
    auto pb = get_pb(p);
    reindex_relocate(pb, p);
    p = pb;
}
return result;
```
`erase` would be implemented like `return do_erase(pos, [this](iterator) { return end(); })`
`stable_erase` would be implemented like `return do_erase(pos, [](iterator p) { return ++p; })`

Reply:
I looked at the same a while back. A subtle difference (easy to miss) is that memcpy is memmov here and acts on multiple elements, instead of just 1. I still feel that it can probably be a shared implementation, but it require some scrutiny.

Reply:
This has been refactored to a common implementation.

Reply:
For the record I was fine with it before the refactoring